### DISCUSSION TITLE
Meet `assertion error assert loss <= 1` when running tools/train.py

### DISCUSSION THREAD

**Discussion by junyizhao04** [PaddlePaddle/PaddleOCR#6864] [Q&A]
打算测试的时候先用一下demo的ICDAR数据集和测试代码，结果直接报错。。。
Traceback (most recent call last):
  File "tools/train.py", line 191, in <module>
    main(config, device, logger, vdl_writer)
  File "tools/train.py", line 164, in main
    program.train(config, train_dataloader, valid_dataloader, device, model,
  File "/home/楼主电脑/PaddleOCR/tools/program.py", line 268, in train
    loss = loss_class(preds, batch)
  File "/home/楼主电脑/.local/lib/python3.8/site-packages/paddle/fluid/dygraph/layers.py", line 930, in __call__
    return self._dygraph_call_func(*inputs, **kwargs)
  File "/home/楼主电脑/.local/lib/python3.8/site-packages/paddle/fluid/dygraph/layers.py", line 915, in _dygraph_call_func
    outputs = self.forward(*inputs, **kwargs)
  File "/home/楼主电脑/PaddleOCR/ppocr/losses/det_db_loss.py", line 65, in forward
    loss_binary_maps = self.dice_loss(binary_maps, label_shrink_map,
  File "/home/楼主电脑/.local/lib/python3.8/site-packages/paddle/fluid/dygraph/layers.py", line 930, in __call__
    return self._dygraph_call_func(*inputs, **kwargs)
  File "/home/楼主电脑/.local/lib/python3.8/site-packages/paddle/fluid/dygraph/layers.py", line 915, in _dygraph_call_func
    outputs = self.forward(*inputs, **kwargs)
  File "/home/楼主电脑/PaddleOCR/ppocr/losses/det_basic_loss.py", line 128, in forward
    **assert loss <= 1**
AssertionError
这是什么情况？如何处理？

**Comment by WenmuZhou** [comment]
Does it appear every time？

**Reply by junyizhao04** [reply]
yes - I tried different conda environments and the problem persists.

**Comment by 4dv3ntur3** [comment]
did you solve it? I had same problem and trying [docker environment](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.5/doc/doc_en/environment_en.md#132-docker-environment-preparation) worked.

**Comment by Pawel1B** [comment]
Anyone has solution that can be done strictly on windows without docker involved?

**Reply by Pawel1B** [reply]
cuda 11.2 with aproperiate package solved problem

**Reply by xReniar** [reply]
@Pawel1B What did you install?

**Reply by Pawel1B** [reply]
check which paddle version supports cuda 11.2

**Reply by xReniar** [reply]
Where can I find the paddle version that suppot cuda 11.2?
I have installed : 
`conda install paddlepaddle-gpu==2.5.1 cudatoolkit=11.2 -c https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/Paddle/ -c conda-forge
`

**Reply by Pawel1B** [reply]
with cuDNN v8.2.1?
also check if this version works
python -m pip install paddlepaddle-gpu==2.5.1.post112 -f https://www.paddlepaddle.org.cn/whl/windows/mkl/avx/stable.html

**Reply by xReniar** [reply]
Yes I am using cuDNN v8.2.1, I will try the new installation

Sorry to ask but is the command for linux or for windows?

**Reply by Pawel1B** [reply]
if u have pip just write without python -m (it is probably windows)

**Reply by xReniar** [reply]
Still does not work

**Reply by Pawel1B** [reply]
try pulling whole thing from github, put your coding file there and if that does not work idk how to help
there if some package is giving you tuble fix it by pytting other version and so on.
